app/uk/gov/hmrc/agentoverseasfrontend/views/application/trading_name.scala.html

```diff
@@ -15,50 +15,51 @@
+@import uk.gov.hmrc.hmrcfrontend.views.Implicits.RichErrorSummary
+@import uk.gov.hmrc.govukfrontend.views.Aliases.Input
+@import uk.gov.hmrc.govukfrontend.views.html.components.GovukInput
+@import uk.gov.hmrc.govukfrontend.views.viewmodels.errormessage.ErrorMessage
+@import uk.gov.hmrc.govukfrontend.views.Implicits.RichInput
 
-@this(mainTemplate: main_template, errorPrefix: error_prefix, errorSummary: ErrorSummary, form: FormWithCSRF)
+@this(
+        mainTemplate: main_template, 
+        errorPrefix: error_prefix, 
+        govukErrorSummary: GovukErrorSummary, 
+        form: FormWithCSRF,
+        govukInput: GovukInput,
+        govukButton: GovukButton)
 
 @(tradingNameForm: Form[String], backLink: Option[String] = None)(implicit messages: Messages, request: Request[_], appConfig: AppConfig)
 
-@mainTemplate(title = errorPrefix(tradingNameForm) + Messages("tradingName.title")) {
-
-    <a href="@backLink.getOrElse(routes.ApplicationController.showContactDetailsForm)" class="link-back">@Messages("button.back")</a>
+@mainTemplate(title = errorPrefix(tradingNameForm) + Messages("tradingName.title"), backLinkHref = Some(backLink.getOrElse(routes.ApplicationController.showContactDetailsForm.url))) {
 
     @if(tradingNameForm.hasErrors) {
-        @errorSummary(Messages("error.summary.heading"),tradingNameForm, Seq.empty, Some("trading-name"))
+        @govukErrorSummary(ErrorSummary().withFormErrorsAsText(tradingNameForm))
     }
 
-    @form(action = routes.ApplicationController.submitTradingName, 'class -> "form js-form") {
-
-        <div class="form-group @if(tradingNameForm.hasErrors){form-field--error soft--ends}">
-           <h1>
-               <label for="tradingName" class="heading-xlarge margin-bottom-20">
-                   <span class="pre-heading">@Messages("tradingName.p1")</span>
-                   @Messages("tradingName.title")
-               </label>
-           </h1>
-            @tradingNameForm.error("tradingName").map { formError => <span class="error-notification" id="tradingName-error">
-                <span class="visuallyhidden">@Messages("error.prefix")</span>
-                @Messages(formError.message)
-            }
-            </span>
-            <input
-                type="text"
-                class="form-control--block"
-                name="tradingName"
-                id="tradingName"
-                spellcheck="false"
-                @tradingNameForm.data.get("tradingName").map { v => value="@v"}
-                @if(tradingNameForm.hasErrors){aria-describedby="tradingName-error" }
-            >
+    <h1 class="govuk-heading-xl">
+        <span class="govuk-caption-xl">@Messages("tradingName.p1")</span>
+        @Messages("tradingName.title")
+    </h1>
 
-        </div>
+    @form(action = routes.ApplicationController.submitTradingName) {
+        @govukInput(Input(
+            spellcheck = Some(false),
+            classes = "govuk-input--width-20",
+            inputType = "text",
```

Comment:
inputType is text by default so you don't need this line

and it looks like the heading on this page should be part of govukInput Label
```
    label = Label(
        isPageHeading = true,
        classes = "govuk-label--xl",
        content = Text(message("key"))
        ),
```